### PR TITLE
Remove constraint in `Stack` to have non-empty list of `transform`s

### PR DESCRIPTION
**Pull request by PierreQuinton**

### CODE REVIEW

File: src/torchjd/autojac/_transform/stack.py [hunk 1]
```diff
@@ -10,15 +10,12 @@
 
 class Stack(Transform[_A, Jacobians]):
```
Comment (PierreQuinton) [inline src/torchjd/autojac/_transform/stack.py:11]: Should it be `Transform[Gradients, Jacobians]` ? We cannot stack anything else than gradients right ?

Comment (ValerianRey) [reply src/torchjd/autojac/_transform/stack.py:11]: I dont think so.

The return type of the contained transforms should always be `Gradients`, but their input type can be anything, as long as its the same for all of them.

Comment (ValerianRey) [reply src/torchjd/autojac/_transform/stack.py:11]: Which is why the type of `transforms` is `transforms: Sequence[Transform[_A, Gradients]]`

Comment (PierreQuinton) [reply src/torchjd/autojac/_transform/stack.py:11]: Yeah of course, my bad

**Comment by ValerianRey** [comment]
LGTM

**Review by ValerianRey — APPROVED** [review]

**Pull request merged**